### ISSUE TITLE
Node Red doesn't load Home Assistant States and errors show in log with 9.1.0

### ISSUE THREAD

**Issue by RogerSelwyn** [hassio-addons/addon-node-red#1023]
# Problem/Motivation

> After upgrade to 9.1.0 errors show in the log and Node Red no longer loads Home Assistant states. See error log below.

## Expected behavior

> Node Red logs show no errors and nodes that access HA data function correctly

## Actual behavior

> Debug errors are shown in Node Red stating entity cannot be found in cache

## Steps to reproduce

> Running correctly with Node Red 9.0.1 add-on, upgrade to 9.1.0, errors are shown and HA access fails
Host reboot is required after snapshot restore to return to correct functioning

## Proposed changes

> n/a

## System Health

version | core-2021.5.5
-- | --
installation_type | Home Assistant Supervised
dev | false
hassio | true
docker | true
virtualenv | false
python_version | 3.8.9
os_name | Linux
os_version | 5.10.17-v7l+
arch | armv7l
timezone | Europe/London

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4944
Installed Version | 1.12.3
Stage | running
Available Repositories | 885
Installed Repositories | 8

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Raspbian GNU/Linux 10 (buster)
-- | --
update_channel | stable
supervisor_version | supervisor-2021.04.3
docker_version | 20.10.6
disk_total | 29.0 GB
disk_used | 11.8 GB
healthy | true
supported | failed to load: Unsupported
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.3.1), SSH & Web Terminal (8.2.1), Node-RED (9.0.1), Dropbox Sync (1.3.0), Portainer (1.5.0), Nginx Proxy Manager (0.11.0), MariaDB (2.3.0)

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
resources | 4
views | 2
mode | storage

</details>

## Error log

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Node-RED
 Flow-based programming for the Internet of Things
-----------------------------------------------------------
 Add-on version: 9.1.0
 You are running the latest version of this add-on.
 System: Raspbian GNU/Linux 10 (buster)  (armv7 / raspberrypi3)
 Home Assistant Core: 2021.5.5
 Home Assistant Supervisor: 2021.04.3
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to WARNING
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] customizations.sh: executing... 
npm WARN deprecated xmldom@0.1.31: Deprecated due to CVE-2021-21366 resolved in 0.5.0
> ejs@2.7.4 postinstall /opt/node_modules/ejs
> node ./postinstall.js
Thank you for installing EJS: built with the Jake JavaScript build tool (https://jakejs.com/)
+ node-red-contrib-ical-events@0.23.0
+ node-red-contrib-gc-trigger@0.0.1
+ node-red-contrib-combine@1.8.1
+ node-red-contrib-advanced-ping@1.2.0
+ luxon@1.27.0
+ rrule@2.6.4
added 21 packages from 41 contributors, updated 1 package and audited 1216 packages in 29.977s
96 packages are looking for funding
  run `npm fund` for details
found 6 vulnerabilities (2 low, 3 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
[cont-init.d] customizations.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] node-red.sh: executing... 
patching file nodes/ui_base.html
Hunk #1 succeeded at 1160 (offset 629 lines).
up to date in 0.79s
found 0 vulnerabilities
[cont-init.d] node-red.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
> addon-node-red@ start /opt
> node $NODE_OPTIONS node_modules/node-red/red.js "--settings" "/etc/node-red/config.js"
20 May 13:16:12 - [warn] Projects disabled : editorTheme.projects.enabled=false
(node:525) UnhandledPromiseRejectionWarning: #<Object>
(Use `node --trace-warnings ...` to show where the warning was created)
(node:525) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 55)
(node:525) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
20 May 13:16:23 - [error] [api-current-state:On ?] Entity could not be found in cache for entity_id: input_boolean.booking
```

**Comment by frenck** [comment]
Most likely going to be fixed with #1021

**Comment by frenck** [comment]
Tonight (5:30PM in my timezone right now).

All build agents have been constantly building today, so it is more a scaling issue today :)

**Comment by frenck** [comment]
Just tagged a new release, should be available any moment now.

Please let me know if this solved the issue (otherwise it probably need to be reported upstream)

**Comment by RogerSelwyn** [comment]
Unfortunately this didn't solve the problem, exactly the same error as before. Strange because my other setup works fine with the 9.1.0 upgrade. That said, it is on Debian not Pi.

**Comment by RogerSelwyn** [comment]
If you need me to gather more info, adding more debug logs or whatever, let me know.

**Comment by garethhowell** [comment]
Just so that @RogerSelwyn knows he's not alone, all my node-red flows stopped with the same errors when the system auto-updated (that'll teach me not to enable auto-update :-)

**Comment by EricE** [comment]
Appear to have the same problem, HassOS on a Nuc.  Used the install snapshot to roll back and I'm back in business.
## System Health

version | core-2021.5.5
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
virtualenv | false
python_version | 3.8.9
os_name | Linux
os_version | 5.4.109
arch | x86_64
timezone | America/Los_Angeles

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4973
Installed Version | 1.12.3
Stage | running
Available Repositories | 822
Installed Repositories | 4

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | May 26, 2021, 5:00 PM
relayer_connected | true
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | false
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 5.13
-- | --
update_channel | stable
supervisor_version | supervisor-2021.04.3
docker_version | 19.03.15
disk_total | 30.8 GB
disk_used | 11.7 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Check Home Assistant configuration (3.7.1), File editor (5.3.1), Samba share (9.4.0), Visual Studio Code (3.4.0), Log Viewer (0.11.0), Node-RED (9.0.1), Terminal & SSH (9.1.2), Mosquitto broker (6.0.1), Portainer (1.5.0), Home Assistant Google Drive Backup (0.104.3), Z-Wave JS (0.1.21), chrony (2.1.0), Glances (0.12.1), ESPHome (1.18.0), InfluxDB (4.1.0), Grafana (6.3.4)

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
resources | 1
views | 4
mode | storage

</details>

**Comment by CervDotBe** [comment]
Got also errors in my log and the node red automations are dead. 

> 22 May 08:37:30 - [info] [server:Home Assistant] Connected to http://supervisor/core
> (node:515) UnhandledPromiseRejectionWarning: #<Object>
> (node:515) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 56)

**Comment by remcoaalbregt** [comment]
I seem to have the same issue with the following setup

## System Health

version | core-2021.5.5
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
virtualenv | false
python_version | 3.8.9
os_name | Linux
os_version | 5.4.83-v8
arch | aarch64
timezone | Europe/Amsterdam

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 5000
Installed Version | 1.11.3
Stage | startup
Available Repositories | 748
Installed Repositories | 5

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 5.13
-- | --
update_channel | stable
supervisor_version | supervisor-2021.04.3
docker_version | 19.03.15
disk_total | 116.7 GB
disk_used | 4.2 GB
healthy | true
supported | true
board | rpi3-64
supervisor_api | ok
version_api | ok
installed_addons | Grafana (6.1.1), InfluxDB (4.0.3), Let's Encrypt (4.11.0), RPC Shutdown (2.2), Samba share (9.3.0), Terminal & SSH (9.1.2), Node-RED (9.1.1), Grocy (0.13.1)

</details>

<details><summary>Lovelace</summary>

dashboards | 2
-- | --
resources | 4
views | 3
mode | storage

</details>

Unfortunately I could not reproduce it in a Docker development environment. After manually downgrading `node-red-contrib-home-assistant-websocket` to `0.30.1` I've managed to resolve the error.

**Comment by gigatexel** [comment]
I have a similar problem after upgrading NodeRed. Unfortunately my latest snapshot with a previous version was just deleted.
Running latest HA, latest NodeRed and latest NodeRed companion.
Some states are no longer available due to connection no longer begin available. Most of the other states are reading just fine.

Is there a way to downgrade to 9.0.1 for example? I tried changing the image using Portainer to 9.0.1, but after a host reboot, 9.1.1 is back.

```24 May 12:42:15 - [error] [function:cfca1e02.5e18] TypeError: Cannot read property 'state' of undefined
24 May 12:42:15 - [error] [ha-entity:Status Boiler Zolder] Sensor update attempted without connection to server.
24 May 12:42:15 - [error] [ha-entity:Status SWW] Sensor update attempted without connection to server.
24 May 12:42:15 - [error] [api-call-service:Boiler aan] Call-Service attempted without connection to server.
24 May 12:42:15 - [error] [function:cfca1e02.5e18] TypeError: Cannot read property 'state' of undefined
24 May 12:42:15 - [error] [function:cfca1e02.5e18] TypeError: Cannot read property 'state' of undefined
24 May 12:42:15 - [error] [ha-entity:Status SWW] Sensor update attempted without connection to server.
24 May 12:42:15 - [error] [function:cfca1e02.5e18] TypeError: Cannot read property 'state' of undefined
24 May 12:42:15 - [error] [ha-entity:Status SWW] Sensor update attempted without connection to server.
24 May 12:42:15 - [error] [ha-entity:Status SWW] Sensor update attempted without connection to server.
```

**Comment by RogerSelwyn** [comment]
Just tried the 9.1.2 upgrade, still the same problem. I note it had upgraded home assistant we socket to 0.31.5

**Comment by zachowj** [comment]
Version 0.31.6 should fix this issue.

**Comment by jrstough** [comment]
Upgrading home assistant web socket to 0.31.6 fixed the issue for me, thanks.

**Comment by Viper-7** [comment]
Still experiencing the same issue with 9.1.2, the Node-Red addon self updated and broke homeassistant integration, and I don't have a recent snapshot. Running the latest stable hassio and homeassistant and no event handling works :/

**Comment by RogerSelwyn** [comment]
9.1.3 seems to have solved it with ha web socket 0.33.0

**Comment by Viper-7** [comment]
Confirmed resolved here with 9.1.3, cheers :)

**Comment by sinclairpaul** [comment]
Closing out then....
